unexpected double retry for local sender

https://circleci.com/gh/tschottdorf/cockroach/583

happened before, but this time we have a complete panic; see the build artifacts in the above link.

```
I0424 21:35:15.425606      97 multiraft.go:696] New Entry[0]: 6/11 EntryNormal [empty]
I0424 21:35:15.425626      97 multiraft.go:699] Committed Entry[0]: 6/11 EntryNormal [empty]
I0424 21:35:15.425850      97 split_test.go:122] split at key "l" complete
I0424 21:35:15.425879      97 split_test.go:116] starting split at key "s"...
I0424 21:35:15.425920      97 range.go:1986] initiating a split of range 5 "l"-"\xff\xff" at key "s"
panic: local sender did not succeed on two attempts: maximum number of attempts exceeded 2

goroutine 234 [running]:
github.com/cockroachdb/cockroach/kv.(*retryableLocalSender).Send(0xc2081469e0, 0x7f41c18a2e70, 0xc2082c5380, 0x7f41c18a3040, 0xc2080a5490, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/local_test_cluster.go:86 +0x400
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).sendOne(0xc20817acd0, 0x7f41c18a2e70, 0xc2082c5380, 0x7f41c18a3040, 0xc2080a5490, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:260 +0x20c
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).Send(0xc20817acd0, 0x7f41c18a2e70, 0xc2082c5380, 0x7f41c18a3040, 0xc2080a5490, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:201 +0xed
github.com/cockroachdb/cockroach/client.(*txnSender).Send(0xc2080d9820, 0x7f41c18a2e70, 0xc2082c5380, 0x7f41c18a3040, 0xc2080a5490, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/client/txn.go:33 +0xd5
--
	/go/src/github.com/cockroachdb/cockroach/kv/local_test_cluster.go:83 +0x2bd
github.com/cockroachdb/cockroach/kv.funcÂ·011()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:142 +0x1fa
created by github.com/cockroachdb/cockroach/kv.(*txnMetadata).close
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:147 +0x88f
FAIL	github.com/cockroachdb/cockroach/kv	0.463s
```

--- a/kv/local_test_cluster.go
+++ b/kv/local_test_cluster.go
@@ -53,12 +53,10 @@ func newRetryableLocalSender(lSender *LocalSender) *retryableLocalSender {
 
 // Send implements the client.Sender interface.
 func (rls *retryableLocalSender) Send(call client.Call) {
-	// Instant retry with max two attempts to handle the case of a
-	// range split, which is exposed here as a RangeKeyMismatchError.
-	// If we fail with two in a row, it's a fatal test error.
+	// Instant retry to handle the case of a range split, which is
+	// exposed here as a RangeKeyMismatchError.
 	retryOpts := util.RetryOptions{
-		Tag:         fmt.Sprintf("routing %s locally", call.Method()),
-		MaxAttempts: 2,
+		Tag: fmt.Sprintf("routing %s locally", call.Method()),
 	}
 	// In local tests, the RPCs are not actually sent over the wire. We
 	// need to clone the Txn in order to avoid unexpected sharing
@@ -82,8 +80,7 @@ func (rls *retryableLocalSender) Send(call client.Call) {
 		return util.RetryBreak, nil
 	})
 	if err != nil {
-		panic(fmt.Sprintf("local sender did not succeed on two attempts: %s",
-			err))
+		panic(fmt.Sprintf("local sender did not succeed: %s", err))
 	}
 }
 

